### ISSUE TITLE
Add further WebLN support

### ISSUE THREAD

**Issue by rolznz** [channelninja/channel.ninja#28]
Hi,
I would like to add some additional WebLN support to channel.ninja to allow users to open and close channels and maybe do other things with WebLN. Since this is a bit more complicated than my first PR, I thought it would be good to open an issue to get thoughts and feedback before I do a proper implementation.

This will also help test the WebLN library and build out better guides for more apps to more easily integrate WebLN.

Ideas:

- Always show my channel info, instead of only when I haven't paid (see 2nd screenshot)
- Fix the channel count (it shows I have 3 channels, when I only seem to have 2 open?)
- Be able to connect to a peer (suggested node)
- Be able to disconnect from a peer (currently this shows on suggested nodes, I guess it should also show in your peers list)
- Be able to open channel with an existing peer
  - What parameters would we like apart from choosing a local and remote balance?
  - How would we enter these parameters, just a simple modal + form for now?
- Be able to close a channel (I need to research how to do this)
- Display more info about your channels, e.g. how much liquidity they have.
- Show pending channel connections? (how do we do this?)

![image](https://user-images.githubusercontent.com/33993199/218358440-6a8232cb-2ac6-466e-af1b-befc748fd05b.png)

![image](https://user-images.githubusercontent.com/33993199/218358605-93bc8050-653d-4f27-ab39-a07da9379f52.png)

What do you think? let me know!

**Comment by channelninja** [comment]
Great ideas! Opening channels via webLN was already on my feature list ;)

I’m on vacation this week. I try to find so time in the next couple days to respond more in detail. 

Nick

**Comment by rolznz** [comment]
Thanks @channelninja - no rush!

Is there a public roadmap or would you consider making one public?

**Comment by rolznz** [comment]
@channelninja I am happy for you to implement it! let me know if you need any help, especially regarding WebLN.

The open channel command for LND has many options - you can see them here: https://lightning.engineering/api-docs/api/lnd/lightning/open-channel-sync

**Comment by rolznz** [comment]
@channelninja I only tested with LND and was able to open a channel (Note: when opening the channel I did have to convert the pubkey from hex to base64). Unfortunately I cannot create a core lightning node using Polar on my windows machine. I would like to try to find a workaround though.

We might be able to test using Alby regtest using the available test nodes here: https://github.com/getAlby/lightning-browser-extension/wiki/Test-setup - I will let you know (hopefully I will have some time to try this weekend)

It looks like the API is available but I haven't added any support to the sandbox app I made for CLN yet.

https://lightning.readthedocs.io/lightning-connect.7.html

**Comment by channelninja** [comment]
@rolznz I implemented the feature and opened a PR https://github.com/channelninja/channel.ninja/pull/48 . 

I couldn't get some of the options from https://lightning.engineering/api-docs/api/lnd/lightning/open-channel-sync to work. When I tried to set a `base_fee` or `fee_rate` alby didn't like it and threw an error. Maybe you have an idea why it's not working?
So for now I just gave the user two settings: channel capacity and funding transaction fee.

Let me know what you think.
